File: c/2017/10/prime.c
/*
 function: prime
 */
#include <stdio.h>
#include <math.h>

int isprime(long n)
{
	long i;
	long m;
	if (n < 6)
		return 0;
	m = (long) sqrt((double)n);

	//printf("number: %ld, sqrt: %ld\n", n, m);
	for (i = 2; i <= m; i++)
		if (n % i == 0)
			return 0;
	return 1;
}

void test(void)
{
	long n;
	int i = 0;

	for (n = 6; n < 1234567; n++)
		if (isprime(n))
			printf("%9ld%c", n, (i++ % 15 == 14) ? '\n' : ' ');
	printf("\ntest OK.\n");
}

int main(void)
{
	test();
	return 0;
}

File: algo/prime.py
#!/usr/bin/env python3
r'''
查找和判断是否素数
'''
import math

def is_prime(n):
    max_factor = int(math.sqrt(n)) + 1 
    # max_factor = n
    for i in range(2, max_factor):
        if n % i == 0:
            # print(n, '=', i , '*', n//i)
            return False
    # print(n, 'is a prime.')
    return True

def list_prime(i):
    ret = []
    for n in range(2, i + 1):
        max_factor = int(math.sqrt(n)) + 1 
        for i in range(2, max_factor):
        #for i in range(2, n):
            if n % i == 0:
                break
        else:
            ret.append(n)
    return ret

def test_list_prime():
    #print(list_prime(100))
    print(list_prime(1_000_000))

if __name__ == '__main__':
    test_list_prime()